storage/gcsemu/meta.go

```diff
@@ -44,13 +56,27 @@ func scrubMeta(meta *storage.Object) {
 	meta.StorageClass = ""
 }
 
-// Return the URL for a file.  Note that this will generally not be useful as an input an http request due to the way
-// that net/url does character substitution (see comments above).
-func bucketUrl(bucket string) string {
-	return fmt.Sprintf("https://www.googleapis.com/storage/v1/b/%s", bucket)
+// Return the URL for a bucket.
+func bucketUrl(baseUrl httpBaseUrl, bucket string) string {
+	if baseUrl == dontNeedUrls || baseUrl == "https://storage.googleapis.com/" {
```

Comment:
I saw the same code repeated twice and maybe you could use a function like:

```
func getUrl(baseUrl httpBaseUrl) {
        if baseUrl == dontNeedUrls || baseUrl == "https://storage.googleapis.com/" {
		baseUrl = "https://www.googleapis.com/"
	} else if baseUrl == "http://storage.googleapis.com/" {
		baseUrl = "http://www.googleapis.com/"
	}

        return baseUrl
}
```

Reply:
Good point